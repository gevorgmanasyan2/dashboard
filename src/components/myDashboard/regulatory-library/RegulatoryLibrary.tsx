import React, { FC, useEffect, useState } from 'react'
import { Tabs } from 'antd'
import usIcon from '../../../assets/icons/us.svg'
import caIcon from '../../../assets/icons/ca.svg'
import wwIcon from '../../../assets/icons/world-wide.svg'
import './regulatoryLibrary.scss'
import RegulatoryLibraryItems from './regulatory-library-items/RegulatoryLibraryItems'
import IsLoading from '../../../spinner/IsLoading'
import { baseAPIURL, config, waiverLibCategoryAPI } from '../../../api/constants'
import  fetch  from 'node-fetch'

interface RegulatoryLibraryProps {
  waiverCategory?: object[]
  isLoading?: boolean
}

type fetchResponse = any

const RegulatoryLibrary: FC<RegulatoryLibraryProps> = () => {
  const [waiverCategoryData, setWaiverCategoryData] = useState()
  const [activeTab, setActiveTab] = useState<string>('1')
  const [load, setLoad] = useState<boolean>(false)
  const { TabPane } = Tabs

  useEffect(() => {
    const apiEndpoint = `${waiverLibCategoryAPI}?where={"country_id":${activeTab}}&expand=1`
    const fetchWaiversCategoryData = async () => {
      setLoad(true)
      const waiverCategoryResponse: fetchResponse = await fetch(`${baseAPIURL}${apiEndpoint}`, config())
      const waiverCategory  = await waiverCategoryResponse.json()
      setWaiverCategoryData(waiverCategory)
      setLoad((load) => !load)
    }
    fetchWaiversCategoryData()
  }, [activeTab])
  const changeTab = (activeKey: string) => {
    setActiveTab(activeKey)
  }
  return (
    <div className="regulatory__library__container">
      <div className="regulatory__library__container__title">
        <h2>Regulatory Resource Library</h2>
      </div>
      <Tabs
        activeKey={activeTab}
        className="regulatory__library__container__tab"
        onChange={changeTab}
      >
        <TabPane key="1" tab={<img src={usIcon} alt="us-flag" />}>
          {load ? <IsLoading /> : <RegulatoryLibraryItems waiverCategory={waiverCategoryData} />}
        </TabPane>
        <TabPane tab={<img src={caIcon} alt="ca-icon" />} key="2">
          {load ? <IsLoading /> : <RegulatoryLibraryItems waiverCategory={waiverCategoryData} />}
        </TabPane>
        <TabPane tab={<img src={wwIcon} alt="ww-icon" />} key="3">
          {load ? <IsLoading /> : <RegulatoryLibraryItems waiverCategory={waiverCategoryData} />}
        </TabPane>
      </Tabs>
    </div>
  )
}

export default RegulatoryLibrary
