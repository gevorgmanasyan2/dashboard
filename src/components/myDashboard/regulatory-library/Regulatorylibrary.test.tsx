/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import RegulatoryLibrary from './RegulatoryLibrary'

it('should render MyDashboard component', () => {
  expect(shallow(<RegulatoryLibrary />)).toMatchSnapshot()
})
