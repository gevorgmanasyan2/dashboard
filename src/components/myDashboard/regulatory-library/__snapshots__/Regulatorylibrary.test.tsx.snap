// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyDashboard component 1`] = `
<div
  className="regulatory__library__container"
>
  <div
    className="regulatory__library__container__title"
  >
    <h2>
      Regulatory Resource Library
    </h2>
  </div>
  <Tabs
    activeKey="1"
    className="regulatory__library__container__tab"
    onChange={[Function]}
  >
    <TabPane
      key="1"
      tab={
        <img
          alt="us-flag"
          src="test-file-stub"
        />
      }
    >
      <RegulatoryLibraryItems />
    </TabPane>
    <TabPane
      key="2"
      tab={
        <img
          alt="ca-icon"
          src="test-file-stub"
        />
      }
    >
      <RegulatoryLibraryItems />
    </TabPane>
    <TabPane
      key="3"
      tab={
        <img
          alt="ww-icon"
          src="test-file-stub"
        />
      }
    >
      <RegulatoryLibraryItems />
    </TabPane>
  </Tabs>
</div>
`;
